packages/contracts-periphery/contracts/universal/op-nft/OptimistInviter.sol

```diff
@@ -0,0 +1,274 @@
+     * @param _signature Signature signed over the claimable invite.
+     */
+    function claimInvite(
+        address _claimer,
+        ClaimableInvite calldata _claimableInvite,
+        bytes memory _signature
+    ) public {
+        // Make sure the claimer and signature have been committed.
+        require(
+            commitments[keccak256(abi.encode(_claimer, _signature))],
+            "OptimistInviter: claimer and signature have not been committed yet"
+        );
+
+        // Generate a EIP712 typed data hash to compare against the signature.
+        bytes32 digest = _hashTypedDataV4(
+            keccak256(
+                abi.encode(
+                    CLAIMABLE_INVITE_TYPEHASH,
+                    _claimableInvite.issuer,
+                    _claimableInvite.nonce
+                )
+            )
+        );
+
+        // Uses SignatureChecker, which supports both regular ECDSA signatures from EOAs as well as
+        // ERC-1271 signatures from contract wallets or multi-sigs. This means that if the issuer
+        // wants to revoke a signature, they can use a smart contract wallet to issue the signature,
+        // then invalidate the signature after issuing it.
+        require(
+            SignatureChecker.isValidSignatureNow(_claimableInvite.issuer, digest, _signature),
+            "OptimistInviter: invalid signature"
+        );
+
+        // The issuer's signature commits to a nonce to prevent replay attacks.
+        // This checks that the nonce has not been used for this issuer before. The nonces are
+        // scoped to the issuer address, so the same nonce can be used by different issuers without
+        // clashing.
+        require(
+            usedNonces[_claimableInvite.issuer][_claimableInvite.nonce] == false,
+            "OptimistInviter: nonce has already been used"
+        );
+
+        // Set the nonce as used for the issuer so that it cannot be replayed.
+        usedNonces[_claimableInvite.issuer][_claimableInvite.nonce] = true;
+
+        // Check the AttestationStation contract to see how many invites the issuer has left.
+        bytes memory attestation = ATTESTATION_STATION.attestations(
+            address(this),
+            _claimableInvite.issuer,
+            CAN_INVITE_ATTESTATION_KEY
+        );
+        // Failing this check means that the issuer was never granted any invites to begin with.
+        require(attestation.length > 0, "OptimistInviter: issuer has no invites");
+
+        uint256 count = abi.decode(attestation, (uint256));
+
+        // Failing this check means that the issuer has used up all of their existing invites.
+        require(count > 0, "OptimistInviter: issuer has no invites");
+
+        // Create the attestation that the claimer can mint from the issuer's invite.
+        // The invite issuer is included in the data of the attestation.
+        ATTESTATION_STATION.attest(
```

Comment:
Although it's technically fine here, I would reduce the count first and then do the external call out of habit because you never know when you'll accidentally introduce a reentrancy bug like this. If the ATST did an external call during this `attest` it could be used to reenter and do multiple transactions without reducing the invite count.

Reply:
nice catch